### PR TITLE
Default limit & upper bound

### PR DESCRIPTION
**Pull request by cammellos**
We allow the user not to specify an upper bound, in which case it will
default to Time.Now() + 60 seconds, to accomodate for messages with
higher timestamp.

Limit is also defaulted to 2000 if not set.

We removed as well the 24 hours limitation as now all the requests will
come with a limit.

### CODE REVIEW

File: mailserver/mailserver.go [hunk 1]
```diff
@@ -535,7 +542,7 @@ func (s *WMailServer) processRequestInBundles(
 		lastEnvelopeHash = envelope.Hash()
 		processedEnvelopes++
 		envelopeSize := whisper.EnvelopeHeaderLength + uint32(len(envelope.Data))
-		limitReached := limit != noLimits && processedEnvelopes == limit
+		limitReached := processedEnvelopes == limit
```
Comment (dshulyak) [inline mailserver/mailserver.go:545]: it doesn't matter if we received less then a limit?

Comment (cammellos) [reply mailserver/mailserver.go:545]: what do you mean exactly?
If there are no more envelopes (less than the limit) we leave the loop, and never reach this condition, is this what you were thinking?

**Comment by dshulyak** [comment]
> We allow the user not to specify an upper bound, in which case it will
default to Time.Now() + 60 seconds, to accomodate for messages with
higher timestamp.

do you suspect that we are missing some messages because sender time is slightly out of sync?

**Review by dshulyak — APPROVED** [review]

**Comment by cammellos** [comment]
@dshulyak it does happen to me at least on the emulator, it gets out of sync fairly easily and I miss recent messages if going from offline->online.

File: mailserver/mailserver.go (hunk 1 above)
Comment (dshulyak) [reply mailserver/mailserver.go:545]: yeah, sorry. i didn't realize that this part was reading from db until limit is reached..

File: mailserver/mailserver.go [hunk 2]
```diff
@@ -40,7 +40,10 @@ import (
 
 const (
 	maxQueryRange = 24 * time.Hour
-	noLimits      = 0
+	defaultLimit  = 2000
+	// When we default the upper limit, we want to extend the range a bit
+	// to accomodate for envelopes with slightly higher timestamp
+	whisperTTLSafeThreshold = 60
```
Comment (adambabik) [inline mailserver/mailserver.go:46]: We should add that it is in seconds.

**Review by adambabik — APPROVED** [review]

**Pull request merged**